### PR TITLE
snap-seccomp: link libseccomp statically to snap-seccomp

### PR DESCRIPTION
**Pull request by mvo5**
This ensures we do not run into library version mismatches when
snap-seccomp is run from the core snap.

This also needs to be pushed to the release/2.26 branch

### CODE REVIEW

File: cmd/snap-seccomp/main.go [hunk 1]
```diff
@@ -19,6 +19,10 @@
 
 package main
 
+//#cgo pkg-config: --static --libs libseccomp
```
Comment (zyga) [inline cmd/snap-seccomp/main.go]: Ironically this is incorrect as you'd actually link with libseccomp *dynamically*. This outputs `-lseccomp` which is added to the compiler. The line below hard-codes that same (effective) library between two linker options that switch link mode. I think we should drop the pkg-config line and just keep the LDFLAGS line below.

I'll test this locally and push if you agree.

File: packaging/ubuntu-14.04/control [hunk 2]
```diff
@@ -21,7 +21,7 @@ Build-Depends: autoconf,
                init-system-helpers,
                libapparmor-dev,
                libglib2.0-dev,
-               libseccomp-dev (>= 2.1.1-1ubuntu1~trusty3),
+               libseccomp-dev (>= 2.1.1-1ubuntu1~trusty4),
```
Comment (zyga) [inline packaging/ubuntu-14.04/control]: What is this update about?

Comment (mvo5) [inline packaging/ubuntu-14.04/control]: libseccomp-dev in trusty does not have a static library. I added it for trusty and uploaded a new SRU to unblock this, see https://launchpad.net/ubuntu/+source/libseccomp/2.1.1-1ubuntu1~trusty4

File: cmd/snap-seccomp/main.go (hunk 1 above)
Comment (mvo5) [inline cmd/snap-seccomp/main.go]: Sure, that works for me

File: tests/main/snap-seccomp/task.yaml [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
-        exit 0
-    fi
-
     echo "Install test-snapd-tools and verify it works"
     snap install test-snapd-tools
     test-snapd-tools.echo hello | MATCH hello
 
+    # FIXME: use dirs.sh in 2.27+
+    echo "Ensure snap-seccomp is statically linked"
+    if ldd /usr/lib/snapd/snap-seccomp | MATCH libseccomp ; then
```
Comment (niemeyer) [inline tests/main/snap-seccomp/task.yaml]: Nice, thanks for testing this.

File: cmd/snap-seccomp/main.go (hunk 1 above)
Comment (zyga) [inline cmd/snap-seccomp/main.go]: Pushed, +1 and I'll merge it when green

File: packaging/ubuntu-14.04/control (hunk 2 above)
Comment (zyga) [inline packaging/ubuntu-14.04/control]: :heart: thank you!

**Pull request merged**